crates/ruff_workspace/src/options.rs

```diff
@@ -1370,6 +1373,30 @@ pub struct Flake8ImportConventionsOptions {
     pub banned_from: Option<FxHashSet<String>>,
 }
 
+fn deserialize_and_validate_aliases<'de, D>(
+    deserializer: D,
+) -> Result<Option<FxHashMap<String, String>>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let Some(aliases) = Option::<FxHashMap<String, String>>::deserialize(deserializer)? else {
+        return Ok(None);
+    };
+
+    for (module, alias) in &aliases {
+        if module.is_empty()
+            || module.split('.').any(|part| !is_identifier(part))
+            || !is_identifier(alias)
+        {
+            return Err(de::Error::custom(
+                "Module must be valid identifier separated by single periods and alias must be valid identifier",
+            ));
+        }
```

Comment:
I suggest that we use two dedicated errors: One for `alias` and one for `module`.

Reply:
Done!